k8_cortx_cloud/deploy-cortx-cloud.sh

```diff
@@ -571,312 +572,152 @@ function deleteStaleAutoGenFolders()
+        --set "clusterStorageSets.${storage_set_name}.durability.sns=${storage_set_dur_sns}"
+        --set "clusterStorageSets.${storage_set_name}.durability.dix=${storage_set_dur_dix}"
+        --set "clusterStorageSets.${storage_set_name}.controlUuid=$(< "${auto_gen_control_path}/id")"
+        --set "clusterStorageSets.${storage_set_name}.haUuid=$(< "${auto_gen_ha_path}/id")"
+    )
+    for node in "${node_name_list[@]}"; do
+        helm_install_args+=(
+            --set "clusterStorageSets.${storage_set_name}.nodes.${node}.dataUuid=$(< "${cfgmap_path}/auto-gen-${node}-${namespace}/data/id")"
+            --set "clusterStorageSets.${storage_set_name}.nodes.${node}.serverUuid=$(< "${cfgmap_path}/auto-gen-${node}-${namespace}/server/id")"
+        )
+        if ((num_motr_client > 0)); then
+            helm_install_args+=(
+                --set "clusterStorageSets.${storage_set_name}.nodes.${node}.clientUuid=$(< "${cfgmap_path}/auto-gen-${node}-${namespace}/client/id")"
+            )
         fi
     done
-    echo $kubectl_cmd_output
 
-    # Create server machine ID config maps
-    for i in "${!node_name_list[@]}"; do
-        auto_gen_cfgmap_path="$cfgmap_path/auto-gen-${node_name_list[i]}-$namespace/server"
-        kubectl_cmd_output=$(kubectl create configmap "cortx-server-machine-id-cfgmap-${node_name_list[i]}-$namespace" \
-                            --namespace=$namespace \
-                            --from-file=$auto_gen_cfgmap_path)
-        if [[ "$kubectl_cmd_output" == *"no such file or directory"* ]]; then
-            printf "Exit early. Create config map 'cortx-server-machine-id-cfgmap-${node_name_list[i]}-$namespace' failed with error:\n$kubectl_cmd_output\n"
-            exit 1
+    # Populate the cluster storage volumes
+    for cvg_index in "${cvg_index_list[@]}"; do
+        cvg_key="solution.storage.${cvg_index}"
+        cvg_name="$(parseSolution "${cvg_key}.name" | cut -f2 -d'>' || true)"
+        cvg_type="$(parseSolution "${cvg_key}.type" | cut -f2 -d'>' || true)"
+        cvg_metadata_device=$(parseSolution "${cvg_key}.devices.metadata.device" | cut -f2 -d'>' || true)
+
+        helm_install_args+=(
+            --set "clusterStorageVolumes.${cvg_name}.type=${cvg_type}"
+            --set "clusterStorageVolumes.${cvg_name}.metadataDevices[0]=${cvg_metadata_device}"
+        )
+
+        IFS=';' read -r -a cvg_dev_var_val_array <<< "$(parseSolution "solution.storage.${cvg_index}.devices.data.d*.device" || true)"
+        for idx in "${!cvg_dev_var_val_array[@]}"; do
+            cvg_dev=$(echo "${cvg_dev_var_val_array[idx]}" | cut -f2 -d'>')
+            helm_install_args+=(
+                --set "clusterStorageVolumes.${cvg_name}.dataDevices[${idx}]=${cvg_dev}"
+            )
+        done
+    done
+
+    helm install \
+        "cortx-cfgmap-${namespace}" \
+        cortx-cloud-helm-pkg/cortx-configmap \
+        --set fullnameOverride="cortx-cfgmap-${namespace}" \
+        "${helm_install_args[@]}"
```

Comment:
Once concern here is that the list of `--set` arguments may get very large (.e.g. hundreds of disks, etc.). There is a limit to the size of the command line arguments. On the systems I'm using, that's 2MB, it could vary.

Reply:
We can investigate https://helm.sh/docs/helm/helm_install/#helm-install options and look at supplying the node/disk layouts via a dynamically generated values.yaml file or if using the `--set-file` flag works to get around the 2MB limit. Using a dynamically generated values.yaml file is still a valid improvement and implementation, as it passes all the information directly to Helm still and not pointers to other locally generated files (ie `helm get values {release}` will get you the entirety of the values we need to deploy the chart, not pointers to temporary files that were previously created during a deploy run).

Reply:
Agreed, YAML values files are a reasonable alternative and that was actually the approach I considered first. The downside is we need to generate them in bash. 😅